modules/nextflow/src/main/groovy/nextflow/k8s/client/K8sClient.groovy

```diff
@@ -244,14 +244,16 @@ class K8sClient {
             return state
         }
 
-        if( status?.phase == 'Pending' && status.conditions instanceof List ) {
-            final allConditions = status.conditions as List<Map>
-            final cond = allConditions.find { cond -> cond.type == 'PodScheduled' }
-            if( cond.reason == 'Unschedulable' ) {
-                def message = "K8s pod cannot be scheduled"
-                if( cond.message ) message += " -- $cond.message"
-                //def cause = new K8sResponseException(resp)
-                log.warn1(message)
+        if( status?.phase == 'Pending' ){
+            if( status.conditions instanceof List ) {
+                final allConditions = status.conditions as List<Map>
+                final cond = allConditions.find { cond -> cond.type == 'PodScheduled' }
+                if( cond.reason == 'Unschedulable' ) {
+                    def message = "K8s pod cannot be scheduled"
+                    if( cond.message ) message += " -- $cond.message"
+                    //def cause = new K8sResponseException(resp)
+                    log.warn1(message)
+                }
             }
```

Comment:
I'm not understanding why splitting the `if` should be different.

Reply:
@pditommaso - You need a little more of the code that's surrounding this to see why than github is showing in this pull request.

The issue is that Kubernetes will sometimes return the state of a Pod so that phase == 'Pending', but it returns a null value for the conditions since there are not conditions yet (the conditions key is not set in the Map returning a null).  Kubernetes will eventually return the conditions List as soon as there are some conditions in it.

Before this pull request:

```
if( status?.phase == 'Pending' && status.conditions instanceof List ) {
            final allConditions = status.conditions as List<Map>
            final cond = allConditions.find { cond -> cond.type == 'PodScheduled' }
            if( cond.reason == 'Unschedulable' ) {
                def message = "K8s pod cannot be scheduled"
                if( cond.message ) message += " -- $cond.message"
                //def cause = new K8sResponseException(resp)
                log.warn1(message)
            }
            // undetermined status -- return an empty response
            return Collections.emptyMap()
        }

        throw new K8sResponseException("K8s invalid pod status (missing container status)", resp)
    }
```

That is, when this issue crops up (the phase is Pending, but there are no conditions yet (a null value instead of List,) then Nextflow will skip all of the if statement logic and throw a K8SReponseException aborting the run.

After this pull request:

```
if( status?.phase == 'Pending' ){
            if( status.conditions instanceof List ) {
                final allConditions = status.conditions as List<Map>
                final cond = allConditions.find { cond -> cond.type == 'PodScheduled' }
                if( cond.reason == 'Unschedulable' ) {
                    def message = "K8s pod cannot be scheduled"
                    if( cond.message ) message += " -- $cond.message"
                    //def cause = new K8sResponseException(resp)
                    log.warn1(message)
                }
            }
            // undetermined status -- return an empty response
            return Collections.emptyMap()
        }

        throw new K8sResponseException("K8s invalid pod status (missing container status)", resp)
```

That is, if phase is Pending, but there are no conditions, then we return an undetermined status.

Only if we reach the end of the procedure and haven't been able to determine the status do we return an K8sResponseException.

I see that I've not changed the message the final K8sResponseException contains so that it's midleading.  I'll fix that and update this pull request with it

Reply:
Got it thanks